### PR TITLE
CMake: fix custom target: directories are no longer used as inputs

### PR DESCRIPTION
**Pull request by mensinda**

### CODE REVIEW

File: mesonbuild/cmake/interpreter.py [hunk 1]
```diff
@@ -714,7 +714,7 @@ def ensure_absolute(x: Path) -> Path:
             # targets, etc. This reduces the chance of misdetecting input files
             # as outputs from other targets.
             # See https://github.com/mesonbuild/meson/issues/6632
-            if not raw.is_absolute() and (self.current_src_dir / raw).exists():
+            if not raw.is_absolute() and (self.current_src_dir / raw).exists() and (self.current_src_dir / raw).is_file():
```
Comment (eli-schwartz) [inline mesonbuild/cmake/interpreter.py:717]:
```suggestion
            if not raw.is_absolute() and (self.current_src_dir / raw).is_file():
```

We don't need to check both.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: mesonbuild/cmake/interpreter.py (hunk 1 above)
Comment (eli-schwartz) [reply mesonbuild/cmake/interpreter.py:717]: ping?

**Comment by jpakkane** [comment]
Ping indeed, this is the last MR needed for 0.62.2.

File: mesonbuild/cmake/interpreter.py (hunk 1 above)
Comment (mensinda) [reply mesonbuild/cmake/interpreter.py]: Forgot about that one...

**Pull request merged**